### PR TITLE
Some `MockWebSocketServer` tweaks

### PR DESCRIPTION
**Pull request by nvie**
Betters internal names, better public names for APIs (matching other WebSocket Server implementations), better comments, and… a bug fix.

### CODE REVIEW

File: packages/liveblocks-core/src/__tests__/_utils.ts [hunk 1]
```diff
@@ -141,58 +146,46 @@ export class MockWebSocketServer {
...
     // Run the callback in the next tick. This is important, because we first
     // need to return the socket.
     setTimeout(() => {
-      this.newSocketCallback?.(socket);
+      // Call the provided callback once, for this connection only.
+      initFn?.(clientSocket);
+
+      // ...then proceed to call the rest of the callbacks, which will be
+      // executed on every new connection
+      this.newConnectionCallbacks.notify({
```
Comment (jrowny) [inline packages/liveblocks-core/src/__tests__/_utils.ts:178]: how does this get populated? It looks like the previous code wrapped the callback everytime it was set.

**Review by jrowny — APPROVED** [review]
looks good

File: packages/liveblocks-core/src/__tests__/_utils.ts (hunk 1 above)
Comment (nvie) [reply packages/liveblocks-core/src/__tests__/_utils.ts:178]: Correct, and that was a bug. It gets populated by calling `wss.onConnection()`. In a unit test, you can use this to program how the server will behave once a connection gets made. For example:

Screen Shot 2023-05-26 at 01 15 23@2x

Comment (nvie) [reply packages/liveblocks-core/src/__tests__/_utils.ts:178]: The point of stubbing out this backend like in this example is to let us interact with and observe effects from the outside. Compare this rewritten test above to the same test as it currently exists on `main`, which fakes the behavior to test here by not using a socket connection at all, but instead making equivalent calls to room's internals.

https://github.com/liveblocks/liveblocks/blob/e0bd5a582370d2bec2b1e68e4873fa00a7332cd5/packages/liveblocks-core/src/__tests__/room.test.ts#L641-L661

**Pull request merged**